cmd/server/main.go

```diff
@@ -1,16 +1,48 @@
 package main
 
 import (
+	"github.com/fasdalf/train-go-musthave-metrics/internal/common/jsonofflinestorage"
 	"github.com/fasdalf/train-go-musthave-metrics/internal/common/metricstorage"
 	"github.com/fasdalf/train-go-musthave-metrics/internal/server"
 	"github.com/fasdalf/train-go-musthave-metrics/internal/server/config"
+	"log/slog"
+	"net/http"
+	"os"
+	"os/signal"
 )
 
 func main() {
-	memStorage := metricstorage.NewMemStorage()
-	httpServer := server.NewHTTPEngine(memStorage)
-	err := httpServer.Run(config.GetConfig().Addr)
-	if err != nil {
+	c := config.GetConfig()
+	memStorage := metricstorage.NewMemStorageWithSave()
+	fileStorage := jsonofflinestorage.NewJSONFileStorage(memStorage, c.StorageFileName, c.StorageFileRestore, c.StorageFileStoreInterval)
```

Comment:
Если хочется убедиться в том, что метод `Save(...)` вызовется при выходе из main, можно воспользоваться `defer`.

```suggestion
	fileStorage := jsonofflinestorage.NewJSONFileStorage(memStorage, c.StorageFileName, c.StorageFileRestore, c.StorageFileStoreInterval)
	defer fileStorage.Save()
```

Reply:
Сигнал всё равно надо перехватывать и останавливать ХТТП сервер, иначе defer не отрабатывает.